### PR TITLE
Rebase feature/file_based_seccomp from master

### PR DESCRIPTION
**Pull request by alindima**
# Reason for This PR

Update file_based_seccomp feature branch to account for latest changes in master.
the feature branch is currently up-to-date with master.

## Description of Changes

This PR adds again the commits merged in #2219, adapting them to fix conflicts.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes are reflected in `firecracker/swagger.yaml`.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

### CODE REVIEW

**Review by luminitavoicu — APPROVED** [review]
LGTM

File: src/vmm/tests/integration_tests.rs [hunk 1]
```diff
@@ -369,7 +330,7 @@ fn verify_load_snapshot(snapshot_file: TempFile, memory_file: TempFile) {
                 microvm_state,
                 mem,
                 false,
-                &empty_seccomp_filter,
+                &mut empty_seccomp_filters,
```
Comment (iulianbarbu) [inline src/vmm/tests/integration_tests.rs:333]: Wouldn't restoring the uVM from snapshot reinstall the same seccomp filters?
Why do we need to pass an empty seccomp filters here?

File: tests/integration_tests/security/demo_seccomp/src/bin/demo_jailer.rs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (iulianbarbu) [inline tests/integration_tests/security/demo_seccomp/src/bin/demo_jailer.rs:1]: Do we want a demo jailer that exerts the default seccomp rules as well?

File: tests/integration_tests/style/test_python.py [hunk 3]
```diff
@@ -27,7 +27,7 @@ def test_python_style():
     python_files = utils.get_files_from(
         find_path="..",
         pattern="*.py",
-        exclude_names=["build"])
+        exclude_names=["build", ".kernel"])
```
Comment (iulianbarbu) [inline tests/integration_tests/style/test_python.py:30]: Not in the scope of the rebase, but I am wondering if we can download the kernel within the container, outside the firecracker workspace directory. Have you tried it?

File: tests/integration_tests/security/demo_seccomp/src/bin/demo_jailer.rs (hunk 2 above)
Comment (alindima) [reply tests/integration_tests/security/demo_seccomp/src/bin/demo_jailer.rs:1]: The demo jailer can take any rules as input. The question then becomes: do we want a test that installs the default seccomp rules.

And this it a tricky question. Esentially, the Firecracker process is a jailer that installs the default seccomp rules.

Do you have any specific idea of a test scenario that would use such a jailer?

**Review by iulianbarbu — APPROVED** [review]

**Pull request merged**